src/entities/user.entity.ts

```diff
@@ -25,11 +29,41 @@ export class User extends Base {
   @IsNotEmpty()
   password!: string;
 
+  @Column({ name: 'confirmation_token', nullable: true })
+  confirmationToken!: string;
+
+  @Column({ name: 'confirmed_at', nullable: true, type: 'timestamp' })
+  confirmedAt!: Date;
+
+  @Column({ name: 'confirmation_sent_at', nullable: true, type: 'timestamp' })
+  confirmationSentAt!: Date;
+
   comparePassword(password: string): boolean {
     return compareSync(password, this.password);
   }
 
+  isConfirmed(): boolean {
+    return this.confirmedAt !== null;
+  }
+
   private hashPassword() {
     this.password = hashSync(this.password, genSaltSync());
   }
+
+  private createConfirmationToken() {
+    this.confirmationToken = new TokenGenerator(
+      256,
+      TokenGenerator.BASE62
+    ).generate();
+  }
+
+  private sendConfirmationEmail() {
+    const mailer = new Mailer();
+    const payload = {
+      name: `${this.firstName} ${this.lastName}`,
+      link: `http://localhost:3000/api/v1/auth/confirm/${this.confirmationToken}`
```

Comment:
we should make this url environment aware.

Reply:
Yes!!